### ISSUE TITLE
inconsistent selection with block cursor

### ISSUE THREAD

**Issue by andrewbanchich** [zed-industries/zed#20121]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

if i set the cursor shape to `block` and the current position is e.g.

```rust
fn |f|oo(self) {}
```

`ctrl-right` will produce

```rust
fn foo|(|self) {}
```

while `ctrl-shift-right` will produce

```rust
fn fo|o|(self) {}
```

the actual selection area for the latter is correct - cutting only removes `foo`. however, that's confusing for me because the behavior of the former command makes me think that the block cursor's current position is its left, but then when you're using it to select something the actual position is its right.

### Environment

Zed: v0.159.7 (Zed)
OS: Linux Wayland ubuntu 24.04
Memory: 30.7 GiB
Architecture: x86_64
GPU: AMD Radeon Graphics (RADV GFX1103_R1) || radv || Mesa 24.0.9-0ubuntu0.2

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by andrewbanchich** [comment]
also, i just noticed that holding `shift-right` will not move the cursor at all the first time.

**Comment by sanbox-irl** [comment]
Yes, the simpler explanation here is that the block cursor is incorrectly placed when multiple things are selected:

<img width="324" alt="Screenshot 2024-12-29 at 3 53 33 PM" src="https://github.com/user-attachments/assets/57e569c3-3b3c-4973-967b-62bbd9893ef4" />

vs the default cursor shape:

<img width="360" alt="Screenshot 2024-12-29 at 3 53 52 PM" src="https://github.com/user-attachments/assets/9bdcbcf1-49e7-4282-8d0e-676973a592ce" />

When we do a `shift-right`, the selection DOES correctly update, but the cursor doesn't actually move in Block mode. You can verify that the selection does update by doing `shift-right` and then `backspace` -- this will delete the character as you'd expect.

**Comment by sanbox-irl** [comment]
This issue should be closed in favor of #18588 which is the same issue but with more conversation on it

**Comment by andrewbanchich** [comment]
@sanbox-irl this is a different issue - this issue is about the visualization of the block. the selection is correct.

the issue you linked is about deleting too many characters. also, it's been closed since that has been fixed and this has not been.

**Comment by andrewbanchich** [comment]
Still happening.

**Comment by andrewbanchich** [comment]
this has been fixed
